python/lsst/meas/deblender/deblend.py

```diff
@@ -450,3 +451,710 @@ def skipParent(self, source, mask):
+            'deblend_hasStrayFlux', type='Flag',
+            doc=('This source was assigned some stray flux'))
+
+        self.log.trace('Added keys to schema: %s', ", ".join(str(x) for x in (
+                    self.nChildKey, self.psfKey, self.psfCenterKey, self.psfFluxKey,
+                    self.tooManyPeaksKey, self.tooBigKey)))
+
+    @pipeBase.timeMethod
+    def run(self, exposures, sources):
+        """Get the psf from each exposure and then run deblend().
+
+        Parameters
+        ----------
+        exposures: dict
+            Keys of the dict are the names of the filters and values are
+            `lsst.afw.image.exposure.exposure.ExposureF`'s.
+            The exposures should be co-added images of the same
+            shape and region of the sky.
+        sources: dict
+            Keys are the names of the filters and the values are
+            `lsst.afw.table.source.source.SourceCatalog`'s, which
+            should be a merged catalog of the sources in each band.
+        
+        Returns
+        -------
+        flux_catalogs: dict or None
+            Keys are the names of the filters and the values are
+            `lsst.afw.table.source.source.SourceCatalog`'s.
+            These are the flux-conserved catalogs with heavy footprints with
+            the image data weighted by the multiband templates.
+            If `self.config.conserveFlux` is `False`, then this item will be None
+        template_catalogs: dict or None
+            Keys are the names of the filters and the values are
+            `lsst.afw.table.source.source.SourceCatalog`'s.
+            These are catalogs with heavy footprints that are the templates
+            created by the multiband templates.
+            If `self.config.saveTemplates` is `False`, then this item will be None
+        """
+        psfs = {B:exp.getPsf() for B, exp in exposures.items()}
+        return self.deblend(exposures, sources, psfs)
+
+    def _getPsfFwhm(self, psf, bbox):
+        return psf.computeShape().getDeterminantRadius() * 2.35
+
+    def _addChild(self, parentId, peak, sources, heavy):
+        """Add a child to a catalog
+
+        This creates a new child in the source catalog,
+        assigning it a parent id, adding a footprint,
+        and setting all appropriate flags based on the
+        deblender result.
+        """
+        assert len(heavy.getPeaks())==1
+        src = sources.addNew()
+        src.assign(heavy.getPeaks()[0], self.peakSchemaMapper)
+        src.setParent(parentId)
+        src.setFootprint(heavy)
+        src.set(self.psfKey, peak.deblendedAsPsf)
+        src.set(self.hasStrayFluxKey, peak.strayFlux is not None)
+        src.set(self.deblendRampedTemplateKey, peak.hasRampedTemplate)
+        src.set(self.deblendPatchedTemplateKey, peak.patched)
+        src.set(self.runtimeKey, 0)
+        return src
+
+    @pipeBase.timeMethod
+    def deblend(self, exposures, sources, psfs, bands=None):
```

Comment:
This function is waaaaaay too long. To the extent that it makes it hard/tiresome to follow what is going on. Please factor out the individual steps.

Reply:
Is that part of our coding standards? I typically think of functions as reusable code that will be needed in multiple places. In this case all of the steps that need reuse have been factored out into other functions. Personally I prefer long functions to having to continually scroll up and down to find out what a function is doing, especially while I'm debugging.

Reply:
Since I couldn't find any rule on it in the guidelines you are off the hook. But while in general agree with your definition, and find it nice for code that I wrote, I find it hard to read with code I didn't write myself and prefer smaller blocks (with descriptive function names).